docs/source/howto/data.rst

```diff
@@ -248,11 +247,247 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
+
+A list of all implemented operators can be found in the ``QueryBuilder`` docstring.
+
+.. _how-to:data:find:relationships:
+
+Relationships
+-------------
+
+It is possible to query for data based on its relationship to another entity in the database.
+Imagine you are not interested in the calculation jobs themselves, but in one of the outputs they create.
+You can build upon your initial query for all ``CalcJobNode``'s in the database using the relationship of the output to the first step in the query:
+
+.. code-block::
+
+    qb = QueryBuilder()
+    qb.append(CalcJobNode, tag='calcjob')
+    qb.append(Int, with_incoming='calcjob')
+
+In the first ``append`` call, we query for all ``CalcJobNode``'s in the database, and *tag* this step with the *unique* identifier ``'calcjob'``.
+Next, we look for all ``Int`` nodes that are an output of the ``CalcJobNode``'s found in the first step, using the ``with_incoming`` relationship argument.
+Since the ``CalcJobNode`` *created* the ``Int`` node, it is an *incoming* node from the ``Int`` node's perspective.
+
+In the context of our query, we are building a *path* consisting of *vertices* (i.e. the entities we query for) connected by *edges* defined by the relationships between them.
+A list of all the relationships you can use for your query, as well as the entities that they connect, can be found in the ``QueryBuilder`` docstring.
+
+.. _how-to:data:find:projections:
+
+Projections
+-----------
+
+When no *projection* is specified, the default behavior of the ``QueryBuilder`` is to project the entities corresponding to the final vertex of the query path.
+However, in many cases we are not interested in the entities themselves, but rather their PK, UUID, *attributes* or some other piece of information stored by the entity.
+This can be achieved using the ``project`` keyword argument:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(CalcJobNode, tag='calcjob')
+    qb.append(Int, with_incoming='calcjob', project='id')
+
+In the above example, executing the query returns all PK's of the ``Int`` nodes which are outputs of all ``CalcJobNode``'s in the database.
+However, you can project more than one piece of information for one vertex by using a list:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(CalcJobNode, tag='calcjob')
+    qb.append(Int, with_incoming='calcjob', project=['id', 'attributes.value'])
+
+Moreover, you can project information for multiple vertices along the query path:
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(CalcJobNode, tag='calcjob', project='uuid)
+    qb.append(Int, with_incoming='calcjob', project=['id', 'attributes.value'])
+
+Asking only for the properties that you are interested in can result in much faster queries.
+If you want the Aiida-ORM instance, add ``'*'`` to your list of projections:
```

Comment:
What exactly does this refer to in this context? The node itself?

Reply:
Yes. I'll try and make this more clear when rewriting the subsection on projections with regard to @sphuber's comment below.